File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1.Tests/Controllers/PersonControllerTest.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class PersonControllerTest
    {
        [TestMethod]
        public void DetailsWithId()
        {
            // Arrange
            var controller = new PersonController();

            // Act
            var result = (ViewResult)controller.Details(33);

            // Assert
            Assert.AreEqual("Details", result.ViewName);
        }


        [TestMethod]
        public void DetailsWithoutId()
        {
            // Arrange
            var controller = new PersonController();

            // Act
            var result = (RedirectToRouteResult)controller.Details(null);

            // Assert
            Assert.AreEqual("Index", result.RouteValues["action"]);
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1.Tests/Controllers/PersonControllerTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> Public Class PersonControllerTest

    <TestMethod()> _
    Public Sub DetailsWithId()
        ' Arrange
        Dim controller As New PersonController()

        ' Act
        Dim result As ViewResult = controller.Details(33)

        ' Assert
        Assert.AreEqual("Details", result.ViewName)
    End Sub

    <TestMethod()> _
    Public Sub DetailsWithoutId()
        ' Arrange
        Dim controller As New PersonController()

        ' Act
        Dim result As RedirectToRouteResult = controller.Details(Nothing)

        ' Assert
        Assert.AreEqual("Index", result.RouteValues("action"))
    End Sub

End Class
